scenes/globals/pause.gd

```diff
@@ -0,0 +1,114 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME
+	# For now is just these two but in the future we could have: Physics, Enemies, AI, etc
+}
+
+# Private vars
+var _pause_requests: Array[PauseRequest] = []
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
+
+
+# Public interface
+func pause_system(system: System, node: Node) -> void:
+	_add_request(system, node)
+
+
+func unpause_system(system: System, node: Node) -> void:
+	_clear_request(system, node)
+
+	_refresh_systems()
+
+
+func is_paused(system: System) -> bool:
+	return system in paused_systems()
+
+
+func paused_systems() -> Array:
+	return _paused_systems
+
+
+func unpaused_systems() -> Array:
+	return System.values().filter(func(system: System) -> bool: return !is_paused(system))
+
+
+# Private methods
```

Comment:
I don't think this comment is needed, as by convention all private methods start like `_this`. Note to myself to start differentating private and public methods in my own PRs.

Reply:
I thought that since this class in particular has many different methods, it was necessary to clearly differentiate (and emphasize) public vs private. But can remove the comment if you want!